src/composables/useFormatDateTime.ts

```diff
@@ -3,125 +3,173 @@
-
-	const options = ref({
-		format: {
-			timeStyle: 'medium',
-			dateStyle: 'short',
-		} as Intl.DateTimeFormatOptions,
-		relativeTime: 'long' as const,
-		ignoreSeconds: false,
-		...toValue(opts),
-	})
-	const wrappedOptions = computed<Required<FormatDateOptions>>(() => ({ ...toValue(opts), ...options.value }))
+export function useFormatRelativeTime(
+	timestamp: MaybeRefOrGetter<Date | number> = Date.now(),
+	opts: MaybeRefOrGetter<FormatRelativeTimeOptions> = {},
+): Readonly<Ref<string>> {
+	let timeoutId
 
-	/** ECMA Date object of the timestamp */
+	/**
+	 * ECMA Date object of the timestamp
+	 */
 	const date = computed(() => new Date(toValue(timestamp)))
 
-	const formattedFullTime = computed<string>(() => {
-		const formatter = new Intl.DateTimeFormat(getCanonicalLocale(), wrappedOptions.value.format)
-		return formatter.format(date.value)
+	/**
+	 * Reactive options for `formatRelativeTime` method
+	 */
+	const options = computed<FormatDateOptions>(() => {
+		const { language, relativeTime, ignoreSeconds } = toValue(opts)
+		return {
+			...language && { language },
+			...relativeTime && { relativeTime },
+			ignoreSeconds: ignoreSeconds
+				? FEW_SECONDS_AGO[relativeTime || 'long']
+				: false,
+		}
 	})
 
-	/** Time string formatted for main text */
-	const formattedTime = computed<string>(() => {
-		if (wrappedOptions.value.relativeTime !== false) {
-			const formatter = new Intl.RelativeTimeFormat(getLanguage(), { numeric: 'auto', style: wrappedOptions.value.relativeTime })
-
-			const diff = date.value.getTime() - currentTime.value
-			const seconds = diff / 1000
-			if (Math.abs(seconds) <= 90) {
-				if (wrappedOptions.value.ignoreSeconds) {
-					return FEW_SECONDS_AGO[wrappedOptions.value.relativeTime]
-				} else {
-					return formatter.format(Math.round(seconds), 'second')
-				}
-			}
-			const minutes = seconds / 60
-			if (Math.abs(minutes) <= 90) {
-				return formatter.format(Math.round(minutes), 'minute')
-			}
-			const hours = minutes / 60
-			if (Math.abs(hours) <= 24) {
-				return formatter.format(Math.round(hours), 'hour')
-			}
-			const days = hours / 24
-			if (Math.abs(days) <= 6) {
-				return formatter.format(Math.round(days), 'day')
-			}
-			const weeks = days / 7
-			if (Math.abs(weeks) <= 4) {
-				return formatter.format(Math.round(weeks), 'week')
-			}
-			const months = days / 30
-			if (Math.abs(months) <= 12) {
-				return formatter.format(Math.round(months), 'month')
-			}
-			return formatter.format(Math.round(days / 365), 'year')
+	/**
+	 * The formatted relative time
+	 */
+	const relativeTime = ref('')
+	watchEffect(() => updateRelativeTime())
+
+	/**
+	 * Update the relative time string.
+	 * This is the callback for the interval.
+	 */
+	function updateRelativeTime() {
+		relativeTime.value = formatRelativeTime(date.value, options.value)
+
+		if (toValue(opts).update !== false) {
+			timeoutId = window.setTimeout(updateRelativeTime, options.value.ignoreSeconds ? 30000 : 1000)
```

Comment:
Maybe make the interval more adaptive? E.g. if the time is far in the past already, nothing changes in hours and there is no need to check every 30 seconds.